### ISSUE TITLE
Verify no sources/transforms/sinks block graceful shutdown

### ISSUE THREAD

**Issue by LucioFranco** [vectordotdev/vector#546]
Looks like some of our sources (I think, I think the issue is the ones with `tokio_threadpool::blocking`) are causing the graceful shutdown to timeout. We should investigate where this is happening.

**Comment by LucioFranco** [comment]
Update: stdin source blocks graceful and immediate shutdown due to its use of blocking. 

```
Sep 25 12:21:31.603  INFO vector: Log level "info" is enabled.
Sep 25 12:21:31.604  INFO vector: Loading config. path="../playground/vector.toml"
Sep 25 12:21:31.620  INFO vector: Vector is starting. version="0.4.0-dev" git_version="v0.3.0-320-geff3bf2" released="Wed, 25 Sep 2019 16:20:16 GMT" arch="x86_64"
Sep 25 12:21:31.621  INFO vector::topology: Running healthchecks.
Sep 25 12:21:31.621  INFO vector::topology: Starting source "in"
Sep 25 12:21:31.622  INFO vector::topology: Starting sink "out"
Sep 25 12:21:31.623  INFO healthcheck{name=out}: vector::topology::builder: Healthcheck: Passed.
Sep 25 12:21:31.623  INFO source{name="in"}: vector::sources::stdin: Capturing STDIN
^CSep 25 12:21:49.565  INFO vector: Shutting down.
^CSep 25 12:21:50.884  INFO vector: Shutting down immediately.
^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C
```

**Comment by binarylogic** [comment]
@LucioFranco I think this is high enough priority to warrant doing. I would like to make sure we test for this.

**Comment by binarylogic** [comment]
@LucioFranco can you spec this out and add checkbox requirements for each source so that we know exactly what is going on here.

**Comment by LucioFranco** [comment]
I've gone ahead and checked all the sources and ensured that none of them will block the shutdown. The issue with `stdin` was before we were using a version that expects it not to block on a call to `poll` but it turned out this was not true. Now that we have moved `stdin` to use a background thread this problem is solved.

As for testing this, it gets very complicated because we would need to spin up the source in a topology setup and actually feed data through it. We may want to think in general how we might want to test a full topology setup.

I've opened #1089 to further look into how we can ensure this issue doesn't come up again.
